src/dodal/common/crystal_metadata.py

```diff
@@ -0,0 +1,65 @@
+import math
+from dataclasses import dataclass
+from enum import Enum
+from typing import Literal
+
+
+@dataclass(frozen=True)
+class Material:
+    """
+    Class representing a crystalline material with a specific lattice parameter.
+    """
+
+    name: str
+    lattice_parameter: float  # Lattice parameter in meters
+
+
+class MaterialsEnum(Enum):
+    Si = Material(name="silicon", lattice_parameter=5.4310205e-10)
+    Ge = Material(name="germanium", lattice_parameter=5.6575e-10)
+
+
+@dataclass(frozen=True)
+class CrystalMetadata:
+    """
+    Metadata used in the NeXus format,
+    see https://manual.nexusformat.org/classes/base_classes/NXcrystal.html
+    """
+
+    usage: Literal["Bragg", "Laue"] | None = None
+    type: str | None = None
+    reflection: tuple[int, int, int] | None = None
+    d_spacing: tuple[float, str] | None = None
+
+    def __init__(
+        self,
+        material: MaterialsEnum,
+        reflection_plane: tuple[int, int, int],
+        usage: str = "Bragg",
+    ) -> None:
+        # Determine material from reflection plane prefix
+        # Set attributes using object.__setattr__ since the class is frozen
+        assert all(
+            i > 0 for i in reflection_plane
+        ), "Reflection plane indices must be positive integers"
+        object.__setattr__(self, "type", material.value.name)
+        object.__setattr__(self, "reflection", reflection_plane)
+        d_spacing = self.calculate_d_spacing(
+            material.value.lattice_parameter, reflection_plane
+        )
+        object.__setattr__(self, "d_spacing", d_spacing)
+        object.__setattr__(self, "usage", usage)
+
+    @staticmethod
+    def calculate_d_spacing(
```

Comment:
Should: So now for the i22 version of the DCM we have to calculate the `d-spacing` ourselves but for the i03/i18 version it's on a PV? Could you reach out to the i22 controls guys to get the PV put there and add a comment to https://github.com/DiamondLightSource/dodal/issues/592 that this is a pre-requisite

Reply:
I think remove it, I don't see why anyone would want to regularly calculate it for comparison so it'll just be extra code that's never used.

Reply:
good idea - so both that and the frozen class now need changing